### PR TITLE
Update the documentation and enable -Wall -Werror

### PR DESCRIPTION
**Pull request by tuliom**
This PR has 1 patch to add more documentation, 5 patches fixing warnings pointed out by the compiler and a last patch to enable `-Wall -Werror`.

Notice that, because of issue #56 , there are still warnings pending.

### CODE REVIEW

**Review by rzinsly — APPROVED** [review]
Looks good but I have some suggestions

File: lib/nx_zlib.c [hunk 1]
```diff
@@ -933,8 +933,6 @@ void nx_hw_init(void)
 	nx_config.strm_inf_bufsz = (1<<16); /* affect the inflate fifo_in and fifo_out */
 	nx_config.soft_copy_threshold = 1024; /* choose memcpy or hwcopy */
 	nx_config.compress_threshold = (10*1024); /* collect as much input */
-	nx_config.inflate_fifo_in_len = ((1<<16)*2); /* default 128K, half used */
-	nx_config.inflate_fifo_out_len = ((1<<24)*2); /* default 32M, half used */
```
[the remark this answers is not in the document]
Comment (tuliom) [reply lib/nx_zlib.c:937]: Good point.  With that, we can remove it from:
`lib/nx_zlib.h:118:      int      inflate_fifo_in_len;`

File: inc_nx/nxu.h [hunk 2]
```diff
@@ -362,6 +374,49 @@ typedef struct {
...
+					 * interruption points will yield much
+					 * less unprocessed source.
+					 * This field is also used by software
+					 * to figure out the first partial or
+					 * full byte that wasn’t processed in
+					 * preparation for resuming the
+					 * decompression. The low order 3 bits
+					 * are fed back to the accelerator as
+					 * part of the resume function, along
+					 * with the adjusted source.
```
[the remark this answers is not in the document]
Comment (tuliom) [reply inc_nx/nxu.h:401]: We should have more references indeed.
I'm fixing this.

**Comment by tuliom** [comment]
@rzinsly Besides the 3 issues you found changed:
1. Removed 2 macro definition from config.mk
2. Removed 2 lines from lib/nx_zlib.h
3. Small fixes to indentation and style of inc_nx/nxu.h

**Comment by rzinsly** [comment]
> @rzinsly Besides the 3 issues you found changed:
> 
>     1. Removed 2 macro definition from config.mk

@tuliom I'm not seeing this, did you update the correct branch?

File: config.mk [hunk 3]
```diff
@@ -11,7 +11,7 @@ LD = $(shell which ld)
 
 ## Common flags
 FLG    = -std=gnu11
-SFLAGS = -O3 -fPIC -D_LARGEFILE64_SOURCE=1 -DHAVE_HIDDEN
```
Comment (tuliom) [inline config.mk:14]: @rzinsly This is where the 2 macro definitions were before the removal.

**Review by rzinsly — APPROVED** [review]
LGTM

**Pull request merged**